### ISSUE TITLE
Make stripe API call async (use a background process)

### ISSUE THREAD

**Issue by mikethetike** [big-neon/bn-api#327]

**Comment by Timo614** [comment]
@mikethetike what exactly is involved with making it async?

Do you mean we should make all of checkout async where a payment is created and then polled until it completes? Or make the auth charge flow async and then it waits at the end so the call is synchronous but the underlying calls in the method aren't?

There's a lot of processing which is contingent on the auth succeeding and the capture happens at the very end of the logic where making it async and waiting won't do much to help. If we have the auth happening in the background and error occurs we'd have to then wait for that anyhow so we could trigger a refund on it (which could be async as we don't need to wait on that response) so it voids the auth (otherwise could take a week to be removed from their accounts which may cause a support burden especially if there are a lot of errors happening at some point).

I'd see some value in making checkout async and polling if that's what this refers to though. Would change the frontend checkout logic a bit though and goes beyond just making the Stripe calls async so confirming if that's the intent.

**Comment by Krakaw** [comment]
@Timo614 By using the Stripe webhooks, we can release the client from the request and use (websockets / long polling) to show the status of the transaction.

This would also allow us to add subscriptions later / longer lasting transactions such as bitcoin payments that may take a fair amount of time to complete.

Let's shelf this until after MVP.

**Comment by Krakaw** [comment]
This is still something that could be long polled, we have implemented the IPN's for Globee, so doing them for Stripe shouldn't be out of the question, which would also improve the /cart/checkout response times. This could be especially nice if we added Push Notification to the checkout completion.
@mikethetike is there any reason we _have_ to stick to doing the payment synchronously?
